File: app/src/main/java/com/bestxty/dl/demo/bean/Task.java
package com.bestxty.dl.demo.bean;

/**
 * @author swjjx
 *         Created by swjjx on 2016/12/20.
 */

public class Task {

    public static final int STATE_READY = 1;
    public static final int STATE_DOWNING = 2;
    public static final int STATE_PAUSE = 3;
    public static final int STATE_DONE = 4;

    private String url;
    private String title;
    private String desc;
    private int state = STATE_READY;
    private Object tag;

    public Task() {
    }

    public Task(String title, String desc, String url) {
        this.title = title;
        this.desc = desc;
        this.url = url;
    }

    @Override
    public String toString() {
        return "Task{" +
                "url='" + url + '\'' +
                ", title='" + title + '\'' +
                ", desc='" + desc + '\'' +
                ", state=" + state +
                '}';
    }

    public Object getTag() {
        return tag;
    }

    public void setTag(Object tag) {
        this.tag = tag;
    }

    public int getState() {
        return state;
    }

    public void setState(int state) {
        this.state = state;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDesc() {
        return desc;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }
}

File: app/src/main/java/com/bestxty/sault/demo/bean/Task.kt
package com.bestxty.sault.demo.bean;

import com.bestxty.sault.Sault;

/**
 * @author swjjx
 *         Created by swjjx on 2016/12/20.
 */

class Task {
    companion object {
        const val STATE_READY: Int = 1
        const val STATE_DOWNING: Int = 2
        const val STATE_PAUSE: Int = 3
        const val STATE_DONE: Int = 4
    }

    var url: String? = null
    var title: String? = null
    var desc: String? = null
    var state: Int = STATE_READY
    var tag: Any? = null
    var priority: Sault.Priority? = null

    constructor() {}


    constructor(title: String, desc: String, url: String, priority: Sault.Priority) {
        this.title = title
        this.desc = desc
        this.url = url
        this.priority = priority
    }

    override fun toString(): String {
        return "Task{" +
                "url='" + url + '\'' +
                ", title='" + title + '\'' +
                ", desc='" + desc + '\'' +
                ", state=" + state +
                '}'
    }


}
